src/Hamiltonians/ExtendedHubbardMom1D.jl

```diff
@@ -0,0 +1,117 @@
+    end
+    kr = range(start; step = step, length = M)
+    ks = SVector{M}(kr)
+    kes = SVector{M}(dispersion.(T , kr .+ boundary_condition))
+    return ExtendedHubbardMom1D{typeof(U),M,typeof(address),U,V,T,boundary_condition}(address, ks, kes)
+end
+
+function Base.show(io::IO, h::ExtendedHubbardMom1D)
+    compact_addr = repr(h.address, context=:compact => true) # compact print address
+    print(io, "ExtendedHubbardMom1D($(compact_addr); u=$(h.u), v=$(h.v), t=$(h.t), boundary_condition=$(h.boundary_condition))")
+end
+
+function starting_address(h::ExtendedHubbardMom1D)
+    return h.address
+end
+
+dimension(::ExtendedHubbardMom1D, address) = number_conserving_dimension(address)
+
+LOStructure(::Type{<:ExtendedHubbardMom1D{<:Real}}) = IsHermitian()
+
+Base.getproperty(h::ExtendedHubbardMom1D, s::Symbol) = getproperty(h, Val(s))
+Base.getproperty(h::ExtendedHubbardMom1D, ::Val{:ks}) = getfield(h, :ks)
+Base.getproperty(h::ExtendedHubbardMom1D, ::Val{:kes}) = getfield(h, :kes)
+Base.getproperty(h::ExtendedHubbardMom1D, ::Val{:address}) = getfield(h, :address)
+Base.getproperty(h::ExtendedHubbardMom1D{<:Any,<:Any,<:Any,U}, ::Val{:u}) where {U} = U
+Base.getproperty(h::ExtendedHubbardMom1D{<:Any,<:Any,<:Any,<:Any,V}, ::Val{:v}) where {V} = V
+Base.getproperty(h::ExtendedHubbardMom1D{<:Any,<:Any,<:Any,<:Any,<:Any,T}, ::Val{:t}) where {T} = T
+Base.getproperty(h::ExtendedHubbardMom1D{<:Any,<:Any,<:Any,<:Any,<:Any,<:Any,BOUNDARY_CONDITION}, 
+    ::Val{:boundary_condition}) where {BOUNDARY_CONDITION} = BOUNDARY_CONDITION
+
+ks(h::ExtendedHubbardMom1D) = getfield(h, :ks)
+
+# standard interface function
+function num_offdiagonals(ham::ExtendedHubbardMom1D, address::SingleComponentFockAddress)
+    singlies, doublies = num_singly_doubly_occupied_sites(address)
+    return num_offdiagonals(ham, address, singlies, doublies)
+end
+
+# 4-argument version
+@inline function num_offdiagonals(ham::ExtendedHubbardMom1D, ::SingleComponentFockAddress, singlies, doublies)
+    M = num_modes(ham)
+    return singlies * (singlies - 1) * (M - 2) + doublies * (M - 1)
+end
+
+@inline function diagonal_element(h::ExtendedHubbardMom1D{<:Any,M,A}, address::A) where {M,A<:SingleComponentFockAddress}
+    map = OccupiedModeMap(address)
+    return dot(h.kes, map) + (h.u/ 2M) * momentum_transfer_diagonal(map) 
+        + (h.v/ M) * extended_momentum_transfer_diagonal(map, (2*π)/M)
+end
+
+@inline function diagonal_element(h::ExtendedHubbardMom1D{<:Any,M,A}, address::A) where {M,A<:FermiFS}
+    map = OccupiedModeMap(address)
+    return dot(h.kes, map) + (h.v/ M) * extended_momentum_transfer_diagonal(map, M)
+end
+
+@inline function get_offdiagonal(
+    ham::ExtendedHubbardMom1D{<:Any,M,A}, address::A, chosen, map=OccupiedModeMap(address)
+) where {M,A<:SingleComponentFockAddress}
+    address, onproduct,_,_,q = momentum_transfer_excitation(address, chosen, map)
+    return address, ham.u/(2*M)*onproduct + ((ham.v * cos((q)*((2*π)/M)))/(M))*onproduct
```

Comment:
```suggestion
    return address, ham.u * onproduct / 2M + ham.v * cos(q * 2π / M) * onproduct / M
```
Clean up expressions like these so they're easier to read